compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -293,12 +297,30 @@ std::optional<std::string> dumpStrToDisk(const std::string &payload,
   return {};
 }
 
+bool hasEnding(std::string const &fullString, std::string const &ending) {
+  if (fullString.length() >= ending.length()) {
+    return fullString.compare(fullString.length() - ending.length(),
+                              ending.length(), ending) == 0;
+  }
+  return false;
+}
+
 // Returns either:
 //  -- the output of running the tool, if run without failure, or
 //  -- an empty optional, if the tool fails to run.
-static std::optional<std::string> runTool(
-    const std::string &program, const std::vector<std::string> &args,
+LogicalResult runTool(
```

Comment:
The comment above isn't valid after this change

Reply:
removed